crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -852,10 +874,13 @@ where
             return Ok(status)
         }
 
-        let res = if self.sync.is_pipeline_idle() {
-            // we can only insert new payloads if the pipeline is _not_ running, because it holds
-            // exclusive access to the database
+        let res = if self.sync.is_pipeline_idle() && self.is_prune_idle() {
+            // we can only insert new payloads if the pipeline and the pruner are _not_ running,
+            // because they hold exclusive access to the database
             self.try_insert_new_payload(block)
+        } else if self.is_prune_active() {
+            warn!(target: "consensus::engine", "Pruning is in progress, buffering new payload.");
```

Comment:
should this be a warning?

I think we should keep the else branch and put the log in an if, so we only have one `self.try_buffer_payload(block)`

Reply:
I think we should have this as a warning, yes, so the node operator could see that they should probably lower the pruning interval. It's not as critical as during the FCU, but still thought worth warning about.

Made one else branch, agree.